libs/mrwebrtc/include/interop_api.h

```diff
@@ -80,38 +80,41 @@ struct mrsRemoteVideoTrackConfig;
 struct mrsDataChannelConfig;
 struct mrsDataChannelCallbacks;
 
-/// Opaque handle to a native PeerConnection interop object.
-using mrsPeerConnectionHandle = void*;
+/// Opaque handle to a native interop object.
+using mrsObjectHandle = void*;
+
+/// Opaque handle to a native reference-counted interop object.
+using mrsRefCountedObjectHandle = mrsObjectHandle;
 
-/// Opaque handle to a native MediaTrack interop object.
-using mrsMediaTrackHandle = void*;
+/// Opaque handle to a native PeerConnection interop object.
+using mrsPeerConnectionHandle = mrsRefCountedObjectHandle;
 
 /// Opaque handle to a native Transceiver interop object.
-using mrsTransceiverHandle = void*;
+using mrsTransceiverHandle = mrsObjectHandle;
 
 /// Opaque handle to a native AudioTrackSource interop object.
-using mrsAudioTrackSourceHandle = void*;
+using mrsAudioTrackSourceHandle = mrsRefCountedObjectHandle;
 
 /// Opaque handle to a native VideoTrackSource interop object.
-using mrsVideoTrackSourceHandle = void*;
+using mrsVideoTrackSourceHandle = mrsRefCountedObjectHandle;
 
 /// Opaque handle to a native LocalAudioTrack interop object.
-using mrsLocalAudioTrackHandle = void*;
+using mrsLocalAudioTrackHandle = mrsRefCountedObjectHandle;
 
 /// Opaque handle to a native LocalVideoTrack interop object.
-using mrsLocalVideoTrackHandle = void*;
+using mrsLocalVideoTrackHandle = mrsRefCountedObjectHandle;
 
 /// Opaque handle to a native RemoteAudioTrack interop object.
-using mrsRemoteAudioTrackHandle = void*;
+using mrsRemoteAudioTrackHandle = mrsObjectHandle;
 
 /// Opaque handle to a native RemoteVideoTrack interop object.
-using mrsRemoteVideoTrackHandle = void*;
+using mrsRemoteVideoTrackHandle = mrsObjectHandle;
 
 /// Opaque handle to a native DataChannel interop object.
-using mrsDataChannelHandle = void*;
+using mrsDataChannelHandle = mrsObjectHandle;
 
 /// Opaque handle to a native ExternalVideoTrackSource interop object.
-using mrsExternalVideoTrackSourceHandle = void*;
+using mrsExternalVideoTrackSourceHandle = mrsRefCountedObjectHandle;
```

Comment:
nit: not incredibly useful since they are all `void*` 😄 

A nicer trick would be:
```cpp
#define DEFINE_HANDLE(NAME) \
  struct NAME ## Impl {}; \
  using NAME ## Handle = const NAME ## Impl*

#define DEFINE_HANDLE_DERIVED(NAME, PARENT) \
  struct NAME ## Impl : public PARENT ## Impl{}; \
  using NAME ## Handle = const NAME ## Impl*

DEFINE_HANDLE(mrsObject);
DEFINE_HANDLE_DERIVED(mrsRefCountedObject, mrsObject);
...
```

C# would still pass raw void pointers in a completely unsafe way, but at least any C++ client would be type safe.

Reply:
Yes, I agree but I wanted to avoid too many unrelated refactors on a non-`master` branch. Arguably this is already too much.

Reply:
Yeah, IMHO it's not incredibly useful as it is redundant with the actual class definitions and API comments and might fall out of sync, so I was thinking, if we want to get some good out of this handles definition let's go all the way! But it doesn't harm so can stay as it is.